upup/pkg/fi/cloudup/awstasks/dnszoneassociation.go

```diff
@@ -0,0 +1,119 @@
+package awstasks
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/glog"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/terraform"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+)
+
+//go:generate fitask -type=DNSZoneAssociation
+type DNSZoneAssociation struct {
+	Name    *string
+	DNSZone *DNSZone
+	VPC     *VPC
+}
+
+var _ fi.CompareWithID = &AutoscalingGroup{}
+
+func (e *DNSZoneAssociation) CompareWithID() *string {
+	return e.Name
+}
+
+func (e *DNSZoneAssociation) Find(c *fi.Context) (*DNSZoneAssociation, error) {
```

Comment:
I probably just don't understand the code well enough but this seems a tad odd. Are we expecting nil for both the object and the error? Would it make sense to return slice of DNSZoneAssociation  or  have three return variables given that find() can generally have three different states. 
1) Something went wrong
2) Something was found 
3) Nothing was found